not-so-smart-contracts/solana/sysvar_account_check/README.md

```diff
@@ -0,0 +1,82 @@
+# Missing Sysvar Account Check
+The sysvar (system account) account is often used while validating access control for restricted functions by confirming that the inputted sysvar account by the user matches up with the expected sysvar account. Without this check in place, any user is capable of passing in their own spoofed sysvar account and in turn bypassing any further authentication associated with it, causing potentially disastrous effects.
+
+## Exploit Scenario
+
+Consider the following:
+
+During signature verification, one of the only points of authentication is the signer being checked against the actual expected address. 
+
+One of the only points of validation of the passed in sysvar account is deserializing the instruction and verifying the data length.
+
+Consider the function load_instruction_at. This takes in the inputted sysvar account data, but on its own, fails to check if the inputted sysvar account is the actual one, making the function vulnerable to spoofing.
+In this scenario, spoofing instruction data to match a valid and authorized address would make an easy attack vector to bypass signature checks.
+
+
+## Example: Wormhole Exploit (February 2022)
```

Comment:
I think we can focus on a smaller, but direct, exploit scenario, and keep the wormhole exploit as a reference.

The reason is that while the issue is simple to understand, putting it in the context of wormhole add unnecessary complexity.

Reply:
I've added an example contract and changed the exploit scenario so it's easier to understand - the Wormhole exploit is separate from them for further reading.